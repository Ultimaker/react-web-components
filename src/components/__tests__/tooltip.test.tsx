// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow, mount } from 'enzyme';

// component
import Tooltip from '../tooltip';

describe('The Tooltip component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            tooltipText: 'Tooltip text'
        };
        wrapper = shallow(<Tooltip {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render to north', () => {
        wrapper.setProps({ direction: 'north' });
        expect(wrapper.find('.tooltip-trigger--north')).toHaveLength(1);
        expect(wrapper.find('.tooltip-trigger--south').exists()).toBe(false);
    });

    it('should disable the tooltip', () => {
        wrapper.setProps({ disableTooltip: true });
        expect(wrapper.find('.disabled')).toHaveLength(1);
    });

    it('should show the tooltip when hovered', () => {
        const mountedWrapper = mount(<Tooltip {...props} />);
        mountedWrapper.simulate('mouseEnter');
        expect(mountedWrapper.find('.show')).toHaveLength(1);
    });

    it('should hide the tooltip when no longer hovered', () => {
        const mountedWrapper = mount(<Tooltip {...props} />);
        mountedWrapper.simulate('mouseEnter');
        expect(mountedWrapper.find('.show')).toHaveLength(1);
        mountedWrapper.simulate('mouseLeave');
        expect(mountedWrapper.find('.show').exists()).toBe(false);
    });

    it('should show the tooltip when selected on touch device', () => {
        const mountedWrapper = mount(<Tooltip {...props} />);
        mountedWrapper.simulate('touchStart');
        expect(mountedWrapper.find('.show')).toHaveLength(1);
    });

    it('should hide the tooltip when unselected on touch device', () => {
        const mountedWrapper = mount(<Tooltip {...props} />);
        mountedWrapper.simulate('touchStart');
        expect(mountedWrapper.find('.show')).toHaveLength(1);
        mountedWrapper.simulate('touchEnd');
        expect(mountedWrapper.find('.show').exists()).toBe(false);
    });

    it('should offset the tooltip', () => {
        wrapper.setState({ tooltipOffset: 100 });
        expect(wrapper.find('.tooltip').prop('style').left).toBe(100);
    });
});
