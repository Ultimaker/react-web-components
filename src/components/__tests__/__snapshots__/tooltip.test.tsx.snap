// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tooltip component should render 1`] = `
<div
  className="tooltip-trigger tooltip-trigger--north"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="tooltip"
  >
    <div
      className="text"
    >
      <p
        key="0"
      >
        Tooltip text
      </p>
    </div>
  </div>
</div>
`;
